src/app.ts

```diff
@@ -0,0 +1,82 @@
+import express, { NextFunction, Response } from 'express';
+import cors from 'cors';
+import rateLimit from 'express-rate-limit';
+import * as leetcode from './leetCode';
+import { FetchUserDataRequest } from './types';
+
+const app = express();
+
+const limiter = rateLimit({
+  windowMs: 60 * 60 * 1000, // 1 hour
+  limit: 60,
+  standardHeaders: 'draft-7',
+  legacyHeaders: false,
+  message: 'Too many request from this IP, try again in 1 hour',
+});
+
+app.use(cors()); //enable all CORS request
+app.use(limiter); //limit to all API
+
+app.get('/', (_req, res) => {
+  res.json({
+    routes: {
+      userDetails: {
+        '/:username': 'get your profile Details',
+        '/:username/badges': 'get your badges',
+        '/:username/solved': 'get total number of question you solved',
+        '/:username/contest': 'get your contest details',
+        '/:username/contest/history': 'get all contest history',
+        '/:username/submission': 'get your last 20 submission',
+        '/:username/submission?limit=7': 'get some of your last submission',
+        '/:username/acSubmission': 'get your last 20 accepted submission',
+        '/:username/acSubmission?limit=7':
+          'get some of your last accepted submission',
+        '/:username/calendar': 'get your submission calendar',
+      },
+      problems: {
+        dailyProblem: { '/daily': 'get daily Problem' },
+        singleProblem: { '/select?titleSlug=two-sum': 'get selected Problem' },
+        problemList: {
+          '/problems': 'get list of 20 problems',
+          '/problems?limit=50': 'get list of some problems',
+          '/problems?tags=array+math': 'get list problems on selected topics',
+          '/problems?tags=array+math+string&limit=5':
+            'get list some problems on selected topics',
+        },
+      },
+    },
+  });
+});
+
+//get the daily leetCode problem
+app.get('/daily', leetcode.dailyProblem);
+
+//get the selected question
+app.get('/select', leetcode.selectProblem);
+
+//get list of problems
+app.get('/problems', leetcode.problems);
+
+// Construct options object on all user routes.
+app.use(
+  '/:username*',
+  (req: FetchUserDataRequest, _res: Response, next: NextFunction) => {
+    req.body = {
+      username: req.params.username,
+      limit: req.query.limit,
+    };
+    next();
+  }
+);
```

Comment:
Since all of the username routes take `username` and `limit` I used this middleware function to make them into a single req.body, then in the actual functions you can call req.body and get type safety.

But on a side note, is limit needed for userdata fetch route?

Reply:
limit need for these 2 routes only

/:username/submission?limit=number
/:username/acSubmission?limit=7

Reply:
gotcha! I'll work on updating tomorrow, probably will remove the middleware since its only for two routes that its needed